services/src/test/groovy/org/jfrog/artifactory/client/CargoPackageTypeRepositoryTests.groovy

```diff
@@ -0,0 +1,97 @@
+package org.jfrog.artifactory.client
+
+import org.hamcrest.CoreMatchers
+import org.jfrog.artifactory.client.model.RepositoryType
+import org.jfrog.artifactory.client.model.repository.settings.RepositorySettings
+import org.jfrog.artifactory.client.model.repository.settings.impl.CargoRepositorySettingsImpl
+import org.testng.annotations.BeforeMethod
+import org.testng.annotations.Test
+
+
+class CargoPackageTypeRepositoryTests extends BaseRepositoryTests {
+
+    CargoPackageTypeRepositoryTests() {
+        remoteRepoUrl = "https://index.crates.io"
+    }
+
+    @Override
+
+    RepositorySettings getRepositorySettings(RepositoryType repositoryType) {
+        def settings = new CargoRepositorySettingsImpl()
+
+        settings.with {
+            // remote
+            cargoAnonymousAccess = rnd.nextBoolean()
+            cargoInternalIndex = rnd.nextBoolean()
+            gitRegistryUrl = "https://index.crates.io/"
+        }
+
+        return settings
+    }
+
+    @BeforeMethod
+    protected void setUp() {
+        prepareVirtualRepo = false
+        super.setUp()
+    }
+
+    @Test(groups = "cargoPackageTypeRepo")
+    void testCargoLocalRepo() {
+        artifactory.repositories().create(0, localRepo)
+        def expectedSettings = localRepo.repositorySettings
+
+        def resp = artifactory.repository(localRepo.getKey()).get()
+        assertThat(resp, CoreMatchers.notNullValue())
+        assertThat(resp.repoLayoutRef, CoreMatchers.is(CargoRepositorySettingsImpl.defaultLayout))
+        resp.getRepositorySettings().with {
+            assertThat(packageType, CoreMatchers.is(expectedSettings.getPackageType()))
+            assertThat(repoLayout, CoreMatchers.is(expectedSettings.getRepoLayout()))
+
+
+            // remote
```

Comment:
This comment says "remote", while this function tests the creation of a local repo. Is this a mistake?
The same goes for the `testCargoFederatedRepo` method.